include/sparrow/iterator.hpp

```diff
@@ -480,9 +479,42 @@ namespace sparrow
             return p_iter < rhs.p_iter;
         }
         
-        pointer p_iter = nullptr;
+        iterator_type p_iter = {};
 
         friend class iterator_access;
     };
 
+    /*
+     * @class pointer_iterator
+     * @brief iterator adaptor for pointers
+     *
+     * pointer_iterator gives an iterator API to a pointer.
+     * @tparam T the pointer to adapt
+     */
+    template <class T>
+    class pointer_iterator;
+
+    template <class T>
+    class pointer_iterator<T*> 
+        : public iterator_adaptor<pointer_iterator<T*>, T*, T, std::contiguous_iterator_tag>
+    {
+    public:
+        
+        using self_type = pointer_iterator<T*>;
+        using base_type = iterator_adaptor<self_type, T*, T, std::contiguous_iterator_tag>;
+        using iterator_type = typename base_type::iterator_type;
+        
+        pointer_iterator() = default;
+        explicit pointer_iterator(const iterator_type& p)
+            : base_type(p)
+        {
+        }
+
+        template <class Iter>
+        requires std::same_as<Iter, std::remove_const_t<T>>
+        pointer_iterator(const Iter& iter)
```

Comment:
Is the template type parameter `Iter` used to ease the signature instead of using a `iterator_adaptor`?

Reply:
This allows to build a pointer_iterator<const T*> from a T*. But we're missing the `pointer_iterator<T*>` => `pointer_iterator<const T*>`